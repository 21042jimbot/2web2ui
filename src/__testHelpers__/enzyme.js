export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
