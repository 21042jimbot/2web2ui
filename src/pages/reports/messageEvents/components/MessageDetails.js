import React, { Component } from 'react';

import { Panel, Tooltip } from '@sparkpost/matchbox';
import { LabelledValue, CopyField } from 'src/components';

import _ from 'lodash';

class MessageDetails extends Component {
  static defaultProps = {
    details: {}
  }

  renderDetails = () => {
    const { details, documentation } = this.props;
    const type = details.type;

    return _.keys(_.omit(details, 'formattedDate')).map((key, i) => {
      const value = details[key];
      const helpText = _.get(documentation, [type, key]);
      const label = helpText ? <Tooltip dark content={helpText}>{key}</Tooltip> : key;

      if (typeof value === 'object') {
        return (
          <LabelledValue key={i} label={label}>
            <CopyField hideCopy value={JSON.stringify(value)} />
          </LabelledValue>
        );
      }

      return <LabelledValue key={i} label={label} value={value} />;
    });
  }

  render() {
    const detailsToRender = _.omit(this.props.details, 'formattedDate');

    return (
      <Panel title='Event Details'>
        <Panel.Section>
          { this.renderDetails(detailsToRender) }
        </Panel.Section>
        <Panel.Section>
          <LabelledValue label='Raw Json'>
            <CopyField value={JSON.stringify(detailsToRender)}/>
          </LabelledValue>
        </Panel.Section>
      </Panel>
    );
  }
}

export default MessageDetails;
