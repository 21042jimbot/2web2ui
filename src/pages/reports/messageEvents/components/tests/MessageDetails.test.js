import React from 'react';
import { shallow } from 'enzyme';
import MessageDetails from '../MessageDetails';

describe('MessageDetails Component', () => {

  it('should render with no props', () => {
    const wrapper = shallow(<MessageDetails />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with all props', () => {
    const props = {
      details: {
        key1: 'val1',
        key2: ['an', 'arrray'],
        type: 'bounce'
      },
      documentation: {
        bounce: {
          key1: 'description for key1'
        }
      }
    };

    const wrapper = shallow(<MessageDetails {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
