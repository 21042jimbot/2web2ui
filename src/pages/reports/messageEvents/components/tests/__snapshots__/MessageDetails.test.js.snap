// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageDetails Component should render with all props 1`] = `
<Panel
  title="Event Details"
>
  <Panel.Section>
    <LabelledValue
      key="0"
      label={
        <WithPositioning(Tooltip)
          content="description for key1"
          dark={true}
        >
          key1
        </WithPositioning(Tooltip)>
      }
      value="val1"
    />
    <LabelledValue
      key="1"
      label="key2"
    >
      <CopyField
        hideCopy={true}
        value="[\\"an\\",\\"arrray\\"]"
      />
    </LabelledValue>
    <LabelledValue
      key="2"
      label="type"
      value="bounce"
    />
  </Panel.Section>
  <Panel.Section>
    <LabelledValue
      label="Raw Json"
    >
      <CopyField
        hideCopy={false}
        value="{\\"key1\\":\\"val1\\",\\"key2\\":[\\"an\\",\\"arrray\\"],\\"type\\":\\"bounce\\"}"
      />
    </LabelledValue>
  </Panel.Section>
</Panel>
`;

exports[`MessageDetails Component should render with no props 1`] = `
<Panel
  title="Event Details"
>
  <Panel.Section />
  <Panel.Section>
    <LabelledValue
      label="Raw Json"
    >
      <CopyField
        hideCopy={false}
        value="{}"
      />
    </LabelledValue>
  </Panel.Section>
</Panel>
`;
